src/promptflow/promptflow/executor/_tool_resolver.py

```diff
@@ -107,7 +111,78 @@ def _convert_to_assistant_definition(self, assistant_definition_path: str, input
         file = self._working_dir / assistant_definition_path
         with open(file, "r", encoding="utf-8") as file:
             assistant_definition = load_yaml(file)
-        return AssistantDefinition.deserialize(assistant_definition)
+        assistant_def = AssistantDefinition.deserialize(assistant_definition)
+        self._resolve_assistant_tool(assistant_def)
+        return assistant_def
+
+    def _resolve_assistant_tool(self, assistant_definition: AssistantDefinition):
+        resolved_tools = {}
+        for tool in assistant_definition.tools:
+            if tool["type"] in ("code_interpreter", "retrieval"):
+                resolved_tools[tool["type"]] = AssistantTool(name=tool["type"], openai_definition=tool, func=None)
+            elif tool["type"] == "function":
+                function_tool = self._load_tool_as_function(tool)
+                resolved_tools[function_tool.name] = function_tool
+            else:
+                raise UnsupportedAssistantToolType(
+                    message_format="Unsupported assistant tool type: {tool_type}",
+                    tool_type=tool["type"],
+                    target=ErrorTarget.EXECUTOR,
+                )
+        assistant_definition.tool_invoker = AssistantToolInvoker(resolved_tools)
+
+    def _load_tool_as_function(self, tool: dict):
+        node, predefined_inputs = self._create_node_for_assistant_tool(tool)
+        resolved_tool = self.resolve_tool_by_node(node, convert_input_types=True)
+        func_name = resolved_tool.definition.function
+        definition = self._generate_tool_definition(func_name, resolved_tool.definition.description, predefined_inputs)
+        if resolved_tool.node.inputs:
+            inputs = {name: value.value for name, value in resolved_tool.node.inputs.items()}
+            func = partial(resolved_tool.callable, **inputs)
+        else:
+            func = resolved_tool.callable
+        return AssistantTool(name=func_name, openai_definition=definition, func=func)
+
+    def _create_node_for_assistant_tool(self, tool: dict):
+        predefined_inputs = {}
+        for input_name, value in tool.get("predefined_inputs", {}).items():
+            predefined_inputs[input_name] = InputAssignment.deserialize(value)
+        node = Node(
+            name="assistant_node",
+            tool="assistant_tool",
+            inputs=predefined_inputs,
+            source=ToolSource.deserialize(tool["source"]) if "source" in tool else None,
+            type=ToolType.PYTHON if "tool_type" in tool and tool["tool_type"] == "python" else None,
+        )
+        return node, list(predefined_inputs.keys())
+
+    def _generate_tool_definition(self, func_name: str, description: str, predefined_inputs: list) -> dict:
```

Comment:
Please add more UTs for different examples here.

Reply:
sure, https://msdata.visualstudio.com/Vienna/_workitems/edit/2980105